doc/source/user_guide/data_sources_tutorial.rst

```diff
@@ -0,0 +1,310 @@
+
+**j) Function** :func:`register_namespace() <ansys.dpf.core.data_sources.DataSources.register_namespace>`
+
+When using an operator that requires data from a DataSources, DPF needs to find
+in its code an internal correspondence to this call. This correspondence is given
+by the namespace, the file extension and the operator name: ``namespace::key::operator_name``.
+
+For example, if the results file comes from a MAPDL solver and has an '.rst' extension
+and you want to get the displacement results in this file, DPF code will get the
+correspondence: ``mapdl::rst::displacement``.
+
+So, if you have an extension that is not
+know by DPF you have to define its namespace. This function is mainly used when
+creating your own operators and plugins, or when you have a file with an unknown
+namespace but you know that it corresponds to certain solver.
+
+The accepted namespaces are those that combine the solvers and its extensions in the
+list: :ref:`Supported apps by DPF and their related formats <list_extensions_solvers>`
+
+.. code-block:: python
+    from ansys.dpf import core as dpf
+    # Create the main DataSources object
+    my_data_sources_j = dpf.DataSources()
+    # Define the path where the main result data can be found
+    my_data_sources_j.set_result_file_path(filepath=r'file.extension', key='extension')
+    # Define the namespace for the results in the given path
+    my_data_sources_j.register_namespace(result_key='extension', namespace='namespace')
+
+
+DataSources exploring
+---------------------
+
+You can check some properties that your ``DataSources`` object have by using
+a helper. They are:
+
+**k) Helper** :attr:`result_key <ansys.dpf.core.data_sources.DataSources.result_key>`
+
+Give which file extension was used by your ``DataSources``.
+
+This extension correspond to the given file, either with the :func:`set_result_file_path() <ansys.dpf.core.data_sources.DataSources.set_result_file_path>` function,
+either if you called the class with the file path as an argument
+
+If the file that you set had more than one extension, only the first one will be returned
+
+.. code-block:: python
+
+    from ansys.dpf import core as dpf
+    # Create the DataSources object
+    my_data_sources_k = dpf.DataSources()
+    # Define the path where the main result data can be found
+    my_data_sources_k.set_result_file_path(filepath=r'file.extension', key='extension')
+
+    # Print the result file extension key
+    print(my_data_sources_k.result_key)
+    # 'extension'
+
+**l) Helper** :attr:`result_files <ansys.dpf.core.data_sources.DataSources.result_files>`
+
+Give the list o list of result files contained in the ``DataSources``. It
+returns the file path of those files
+
+- If you use the :py:func:`set_result_file_path() <ansys.dpf.core.data_sources.DataSources.set_result_file_path>` function it will return only the file path given as an argument to this function
```

Comment:
```suggestion
- If you use the :py:func:`set_result_file_path() <ansys.dpf.core.data_sources.DataSources.set_result_file_path>` function, it will return only the file path given as an argument to this function.
```